### DISCUSSION TITLE
Is it possible to spy on multiple requests that matches the cy.intercept() network requests? (Not just the first one)

### DISCUSSION THREAD

**Discussion by thaiching** [cypress-io/cypress#17142] [General]
Hi, 
   When I intercept the following ads requests in my project, this would normally returns a list of identical ad requests that matches the following interceptor.

`cy.intercept('GET', '**/gampad/ads?*').as('gam');`

I would like to spy on each one of the matching ad requests in order to compare one of the query string parameter (correlator) is the same across all the ad requests.

Is it possible to do that in Cypress?

<img width="747" alt="Untitled" src="https://user-images.githubusercontent.com/7057586/123785799-d8fa1f80-d8d0-11eb-86cf-4e1424104f41.png">

**Comment by flotwig** [comment]
The answer to this question is similar to your [other question](https://github.com/cypress-io/cypress/discussions/17299) - `.all` will yield a list of all matched requests. For example:

```ts
cy.get('@gam.all').then(interceptions => {
	for (const interception of interceptions) {
		expect(interception.request.url).to.include('correlator=12345')
	}
})
```

**Reply by ThaiChingChong** [reply]
Thank you @flotwig for your suggestion. Tried your solution and it works like a charm. :)

This `cy.get('alias.all')` is not documented in Cypress doc but I think it's very useful for lots of test scenarios when it comes to spying on network

**Comment by HakeemOkoh** [comment]
**This solution will help you target a specific request using the index from the list returned by cy.get("@alias.all")**

```
cy.get("@alias.all").then((interceptions) => {
    const interception = interceptions[17];
    cy.wrap(interception).as("interception");
    cy.get("@interception").then((interception) => {
      const responseBody = interception.response.body.data;
      // Perform your assertions here
    });
  });
```
